### ISSUE TITLE
EqualsHashCode: False Negative - class overrides equals() but does not override hashCode()

### ISSUE THREAD

**Issue by leemeii** [checkstyle/checkstyle#19690]
I have read check documentation: https://checkstyle.org/checks/coding/equalshashcode.html
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```text
/var/tmp $ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN" "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="EqualsHashCode"/>
  </module>
</module>

/var/tmp $ cat a.java

public class a { 

    @Override
    public boolean equals(Object obj) {
        return true; 
    }

    private static class Inner {
    }

    public static void main(String[] args) {
        a o1 = new a();
        a o2 = new a();
        System.out.println(o1.equals(o2));
    }
}

/var/tmp $ java -jar checkstyle-13.3.0-all.jar -c config.xml a.java
Starting audit...
Audit done.
```
Describe what you expect in detail.

Checkstyle should report an EqualsHashCode violation for class a because it overrides equals() but does not override hashCode().

Actual Behavior:
No EqualsHashCode is reported at all.

**Comment by romani** [comment]
>java -jar checkstyle-*-all.jar -c config.xml a.java

@leemeii , please confirm version you used

**Comment by DakshRJain737** [comment]
I am on it.

**Comment by DakshRJain737** [comment]
@romani  Using same config file and same example I tried to run jar but i found error as expected i.e no false negative

```
cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC "-//Puppy Crawl//DTD Check Configuration 1.3//EN" "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="EqualsHashCode"/>
  </module>
```

```
cat src/Main.java
public class Main {

    @Override
    public boolean equals(Object obj) {
        return true;
    }

    private static class Inner {
    }

    public static void main(String[] args) {
        Main o1 = new Main();
        Main o2 = new Main();
        System.out.println(o1.equals(o2));
    }
```
```
java -jar checkstyle-13.3.0-all.jar -c config.xml src/Main.java
Starting audit...
[ERROR] /home/dakshjain737/checkstyle-test/src/Main.java:3:5: Definition of 'equals()' without corresponding definition of 'hashCode()'. [EqualsHashCode]
Audit done.
Checkstyle ends with 1 errors.
```
Issue must be closed as already gives violations as expected

**Comment by romani** [comment]
@DakshRJain737 , thanks a lot
